monai/deploy/operators/monai_bundle_inference_operator.py

```diff
@@ -0,0 +1,725 @@
+            self._send_output(output_dict[name], name, input_metadata, op_output, context)
+
+    def predict(self, data: Any, *args, **kwargs) -> Union[Image, Any, Tuple[Any, ...], Dict[Any, Any]]:
+        """Predicts output using the inferer."""
+
+        return self._inferer(inputs=data, network=self._model_network, *args, **kwargs)
+
+    def pre_process(self, data: Any, *args, **kwargs) -> Union[Image, Any, Tuple[Any, ...], Dict[Any, Any]]:
+        """Processes the input dictionary with the stored transform sequence `self._preproc`."""
+
+        if is_map_compose(self._preproc):
+            return self._preproc(data)
+        return {k: self._preproc(v) for k, v in data.items()}
+
+    def post_process(self, data: Any, *args, **kwargs) -> Union[Image, Any, Tuple[Any, ...], Dict[Any, Any]]:
+        """Processes the output list/dictionary with the stored transform sequence `self._postproc`.
+
+        The "processed_inputs", in fact the metadata in it, need to be passed in so that the
+        invertible transforms in the post processing can work properly.
+        """
+
+        # Expect the inputs be passed in so that the inversion can work.
+        inputs = kwargs.get(self.kw_preprocessed_inputs, {})
+
+        if is_map_compose(self._postproc):
+            if isinstance(data, (list, tuple)):
+                outputs_dict = dict(zip(data, self._outputs.keys()))
+            elif not isinstance(data, dict):
+                oname = first(self._outputs.keys())
+                outputs_dict = {oname: data}
+            else:
+                outputs_dict = data
+
+            # Need to add back the inputs including metadata as they are needed by the invert transform.
+            outputs_dict.update(inputs)
+            logging.debug(f"Effective output dict keys: {outputs_dict.keys()}")
+            return self._postproc(outputs_dict)
+        else:
+            if isinstance(data, (list, tuple)):
+                return list(map(self._postproc, data))
+
+            return self._postproc(data)
+
+    def _receive_input(self, name: str, op_input: InputContext, context: ExecutionContext):
+        """Extracts the input value for the given input name."""
+
+        # The op_input can have the storage type of IN_MEMORY with the data type being Image or others,
+        # as well as the other type of DISK with data type being DataPath.
+        # The problem is, the op_input object does not have an attribute for the storage type, which
+        # needs to be inferred from data type, with DataPath meaning DISK storage type. The file
+        # content type may be interpreted from the bundle's network input type, but it is indirect
+        # as the op_input is the input for processing transforms, not necessarily directly for the network.
+        in_conf = self._inputs[name]
+        itype = self._get_io_data_type(in_conf)
+        value = op_input.get(name)
+
+        metadata = None
+        if isinstance(value, DataPath):
```

Comment:
Great to see it is compatible with any storage type!

Just wonder if we were also able to make it work with ordinary image files, instead of pickle files.
For example, If the type of value is DataPath (especially for folder path), instead of removing load operator from bundle's transforms, we may be able to pass the file/folder paths to bundle's load transform?

Reply:
Good question! I should have left more comments in the code because I myself thought about this for quite a bit.

- Was considering loading other, and a limited set of, files, e.g. NIfTI/MHD which we already know exactly what to do, but on second thought, decided to clamp down on only pickle file, and will have more discussion on the list of file types to b supported in the future releases (we may end up with reusing the MONAI Core ImageReader impl if need be).
- In my earlier versions of the bundle operator, as well as in the previous release, I actually did customize the LoadImage transform in the pre-process Compose, albeit with the IN_MEM image reader/converter. But now with the bundle API, it makes more sense to treat the loadImage transform opaquely, i.e. either in or out and no need to customize it when `compute` is called for a specific payload. The also will allow us to have a more flexible `loadimage` impl in another operator, e.g. I was considering a multi-modal use case where a JSON file is one of the inputs, beside an image file (see the first point).